lib/Pass/Pass.cpp

```diff
@@ -385,96 +386,108 @@ struct SouperPass : public ModulePass {
+          KV->hIncrBy(GetReplacementLHSString(Cand.BPCs, Cand.PCs,
+                                              Cand.Mapping.LHS, Context),
+                      HField, 1);
+        }
+        if (DynamicProfileAll) {
+          dynamicProfile(F, Cand);
+          Changed = true;
           continue;
-        } else {
-          report_fatal_error("Unable to query solver: " + EC.message() + "\n");
         }
-      }
-      if (!Cand.Mapping.RHS)
-        continue;
+        if (std::error_code EC =
+            S->infer(Cand.BPCs, Cand.PCs, Cand.Mapping.LHS,
+                     Cand.Mapping.RHS, IC, Cand.Type)) {
+          if (EC == std::errc::timed_out ||
+              EC == std::errc::value_too_large) {
+            continue;
+          } else {
+            report_fatal_error("Unable to query solver: " + EC.message() + "\n");
+          }
+        }
+        if (!Cand.Mapping.RHS)
+          continue;
 
-      if (ReplaceCount >= FirstReplace && ReplaceCount <= LastReplace) {
+        if (ReplaceCount >= FirstReplace && ReplaceCount <= LastReplace) {
 
-	Instruction *I = Cand.Origin;
-	assert(Cand.Mapping.LHS->hasOrigin(I));
-	IRBuilder<> Builder(I);
+          Instruction *I = Cand.Origin;
+          assert(Cand.Mapping.LHS->hasOrigin(I));
+          IRBuilder<> Builder(I);
 
-	Value *NewVal = getValue(Cand.Mapping.RHS, I, EBC, DT,
-                               ReplacedValues, Builder, F->getParent());
-	if (!NewVal) {
-	  if (DebugLevel > 1)
-	    errs() << "\"\n; replacement failed\n";
-	  continue;
-	}
+          Value *NewVal = getValue(Cand.Mapping.RHS, I, EBC, DT,
+                                   ReplacedValues, Builder, F->getParent());
+          if (!NewVal) {
+            if (DebugLevel > 1)
+              errs() << "\"\n; replacement failed\n";
+            continue;
+          }
+
+          ReplacedValues[Cand.Mapping.LHS] = NewVal;
+
+          if (DebugLevel > 1) {
+            if (DebugLevel > 2) {
+              errs() << "\nFunction before replacement:\n";
+              F->print(errs());
+            }
+            errs() << "\n";
+            errs() << "; Replacing \"";
+            I->print(errs());
+            errs() << "\"\n; from \"";
+            I->getDebugLoc().print(errs());
+            errs() << "\"\n; with \"";
+            NewVal->print(errs());
+            errs() << "\" in:\n";
+            PrintReplacement(errs(), Cand.BPCs, Cand.PCs, Cand.Mapping);
+            errs() << "\"\n; with \"";
+            NewVal->print(errs());
+            errs() << "\"\n";
+          }
 
-	ReplacedValues[Cand.Mapping.LHS] = NewVal;
+          if (DynamicProfile)
+            dynamicProfile(F, Cand);
+
+          if (Cand.Type == HarvestedFromUser) {
+            I->replaceAllUsesWith(NewVal);
+          } else {
+            llvm::Value::use_iterator UI = I->use_begin(), E = I->use_end();
+            for (; UI != E;) {
+              llvm::Use &U = *UI;
+              ++UI;
```

Comment:
this should go in the "increment" part of the for loop construct, right?

Reply:
Correct, I'll get this fixed.